Lib/test/string_tests.py

```diff
@@ -767,6 +767,40 @@ def test_replace(self):
         self.checkraises(TypeError, 'hello', 'replace', 42, 'h')
         self.checkraises(TypeError, 'hello', 'replace', 'h', 42)
 
+    def test_replacement_on_buffer_boundary(self):
+
+        # gh-127971: Check we don't read past the end of the buffer when a
+        # potential match misses on the last character. Note this will likely
+        # not cause a failure unless ASAN is enabled, and even that may be
+        # dependent on implementation details subject to change.
+        any_3_nonblank_codepoints = '!!!'
+        seven_codepoints = any_3_nonblank_codepoints + ' ' + any_3_nonblank_codepoints
+        a = (' ' * 243) + seven_codepoints + (' ' * 7)
+        b = ' ' * 6 + chr(256)
```

Comment:
```suggestion
    def test_replacement_on_buffer_boundary(self):
        # gh-127971: Check we don't read past the end of the buffer when a
        # potential match misses on the last character. Note this will likely
        # not cause a failure unless ASAN is enabled, and even that may be
        # dependent on implementation details subject to change.

        any_3_nonblank_codepoints = '!!!'
        seven_codepoints = any_3_nonblank_codepoints + ' ' + any_3_nonblank_codepoints
        a = (' ' * 243) + seven_codepoints + (' ' * 7)
        b = ' ' * 6 + chr(256)
```

> Note this will likely not cause a failure unless ASAN is enabled

In this case, make this test an ASAN test only. I would also shorten the comment.